### ISSUE TITLE
Custom JS/CSS for specific service

### ISSUE THREAD

**Issue by patrickcurl** [getferdi/ferdi#83]
I can't stand light interfaces.. they make me want to gouge my eyes out. 

Github, and some of the other services have white backgrounds it'd be nice to be able to 
pull in my styles from stylish to use in Ferdi.

**Comment by Makazzz** [comment]
There is [an issue from few days ago](https://github.com/getferdi/ferdi/issues/71) about implementing Dark Reader and it is already done, but I myself don't find the way to activate it.

**Comment by Gaboris** [comment]
Hi @patrickcurl

This is a copy-paste of a "guide" I made for another person back on the Franz Git. :)

/*
Custom themes are actually supported, but they need to be applied manually since we don't have a built in method to upload them yet.

For example I made my Messenger nice and dark with just a simple CSS I got from Stylish and several people here who figured out how to make it work. :D

Just download this or some other theme you like.
https://userstyles.org/styles/134433/dark-night-facebook-messenger
You might need to edit some themes to work. For example in this one you'd need to remove the first and last two lines of code.
Then you need to add it to Ferdi's Messenger folder.
copy this into your file explorer %AppData%\Ferdi\recipes\messenger\ and save the theme as darkmode.css
Then you open Ferdi's settings, go the Services tab and pick Messenger where you need to enable dark mode.

If it doesn't work right away you might need to fully restart Ferdi and if it's still not working you might need some more help.
*/

**Comment by Makazzz** [comment]
Thanks @Gaboris, but does Dark Reader work the same way or it is automatic?

**Comment by Gaboris** [comment]
THAT, I don't know. It might be because since the update my custom themes were randomly getting disabled so maybe Dark Reader is reverting them again without me knowing.

I'm gonna post an issue about this if I can do some test with it later on.

**Comment by vantezzen** [comment]
Custom CSS and custom JS injection could be something we could think about implementing.

For your use case, you can use the universal Dark Mode in the next update. It uses DarkReader to dynamically create a dark mode for the current service. Alternatively, you can use the darkmode.css method @Gaboris pointed out. You can find darkmode.css files for common services at <https://github.com/MikeDabrowski/franz-themed-recipes>

**Comment by patrickcurl** [comment]
Thanks everyone for the tips. Will apply them when I get a chance!

**Comment by kytwb** [comment]
Could get inspired/copy from rambox for the implementation of JS/CSS injection: https://rambox.pro/#feature-details/js_css

**Comment by vantezzen** [comment]
Also related: [franz#1587](https://github.com/meetfranz/franz/issues/1587)

**Comment by ylluminate** [comment]
This goes beyond appearance / CSS injection. JS injection is very useful for other things such as autologin for Proton Mail. This is how it's done in Rambox:
![image](https://p153.p0.n0.cdn.getcloudapp.com/items/8LuwPbdm/js_injection.png)

Here's the snippet one would use:
```
setTimeout(() => {
    document.body.scrollTop=0;document.getElementById('username').value = 'PUT_USERNAME_HERE';
    document.getElementById('password').value = 'PUT_PASSWORD_HERE';
    document.getElementById('login_btn').click();
}, 1000);

```

This therefore becomes a critical feature for other aspects.  Should this be made into a new issue or can we work off of this one?

**Comment by kytwb** [comment]
We can work off this one, but nobody has started to work on it afaik. Still up for grabs for now, I'm focusing on fixing the known issues before going back on new features.

**Comment by vantezzen** [comment]
This feature will be implemented via #401

**Comment by kytwb** [comment]
Shipped in [v5.4.4-beta.2](https://github.com/getferdi/ferdi/releases/tag/v5.4.4-beta.2).
